### ISSUE TITLE
Docs: Microsoft Visual C++ 2015 Redistributable is required on Windows

### ISSUE THREAD

**Issue by jborbely** [astral-sh/ruff#11503]
**OS**: Windows 10
**ruff**: version 0.4.4 (installed via `pipx`)

Issue
-----
`ruff.exe` silently fails to run from the terminal if Microsoft Visual C++ 2015 Redistributable is not installed.

Running
```console
> ruff --version
```
does not show the version in the terminal nor is an error shown. Complete silence.

Double-clicking the executable, located at `..\pipx\venvs\ruff\Scripts\ruff.exe`, pops-up the following window

![vcruntime140-error](https://github.com/astral-sh/ruff/assets/4319367/7f97100e-2827-4afc-8d9e-5577dbbdb57d)

Fix
---
Install Microsoft Visual C++ 2015 Redistributable, then

```console
> ruff --version
ruff 0.4.4
```

Suggestion
------------
Perhaps adding something to the docs (https://docs.astral.sh/ruff/installation/) for Windows users that Microsoft Visual C++ 2015 Redistributable must be installed. Either provide the download link (https://www.microsoft.com/en-us/download/details.aspx?id=52685) and/or specify the command to run to install it `winget install --exact --id Microsoft.VCRedist.2015+.x64`

**Comment by T-256** [comment]
Currently Windows builds uses `msvc` runtime. we can eaither use different runtime (using mingw), or statically link it into binary [at link time](https://devblogs.microsoft.com/cppblog/msvc-address-sanitizer-one-dll-for-all-runtime-configurations/) which causes larger binary.

**Comment by jborbely** [comment]
Certainly, those additional options would resolve the external VCRUNTIME###.dll dependency issue. I do not have a preference for which option (update docs, use mingw, statically link, ...) the maintainers of `ruff` decide to go with.

**Comment by MichaReiser** [comment]
@BurntSushi what's your experience with statically linking the CRT on windows? Any drawbacks that you've become aware after shipping the change in ripgrep?

https://github.com/BurntSushi/ripgrep/blob/35160a1cdb4c7e2c370e8a7fad6508ff922a33c2/.cargo/config.toml#L4C8-L8

**Comment by BurntSushi** [comment]
> @BurntSushi what's your experience with statically linking the CRT on windows? Any drawbacks that you've become aware after shipping the change in ripgrep?
> 
> https://github.com/BurntSushi/ripgrep/blob/35160a1cdb4c7e2c370e8a7fad6508ff922a33c2/.cargo/config.toml#L4C8-L8

Not that I'm aware of! It's been without incident to the point that I totally forgot that this change even happened. :-)

**Comment by MichaReiser** [comment]
Would someone be interested in contributing the change to Ruff? It requires porting the configuration from ripgrep to ruff and doing a quick analysis of the file size change between the two versions (release mode).

**Comment by T-256** [comment]
Can I take this one?

**Comment by jborbely** [comment]
> Can I take this one?

@T-256 definitely okay with me.

**Comment by charliermarsh** [comment]
Go for it!
